rules/automount-service-account/raw.rego

```diff
@@ -1,23 +1,26 @@
 package armo_builtins
 
-# Fails if user account mount tokens in pod by default
+# Fails if ServiceAccount mount tokens in pod by default and has at least a bind in (Cluster)RoleBindings
 deny [msga]{
     service_accounts := [service_account |  service_account= input[_]; service_account.kind == "ServiceAccount"]
     service_account := service_accounts[_]
+
+    has_service_account_binding(service_account)
+
     result := is_auto_mount(service_account)
-	failed_path := get_failed_path(result)
+    failed_path := get_failed_path(result)
     fixed_path := get_fixed_path(result)
 
     msga := {
-	    "alertMessage": sprintf("the following service account: %v in the following namespace: %v mounts service account tokens in pods by default", [service_account.metadata.name, service_account.metadata.namespace]),
-		"alertScore": 9,
-		"packagename": "armo_builtins",
-		"fixPaths": fixed_path,
-		"failedPaths": failed_path,
-		"alertObject": {
-			"k8sApiObjects": [service_account]
-		}
-	}
+        "alertMessage": sprintf("the following service account: %v in the following namespace: %v mounts service account tokens in pods by default", [service_account.metadata.name, service_account.metadata.namespace]),
+        "alertScore": 9,
+        "packagename": "armo_builtins",
+        "fixPaths": fixed_path,
+        "failedPaths": failed_path,
+        "alertObject": {
+            "k8sApiObjects": [service_account]
+        }
+    }
```

Comment:
looking over this rule again I realise we don't need this deny for the purpose of control C-0261, we only want to return workloads. Plus we want the check of `has_service_account_binding(service_account)` for the service accounts that are being used by those workloads that mount its token. Perhaps we should create a new rule after all which is very similar to this but with those slight changes. Sorry for misleading you

Reply:
Adding that in the new rule we should create a filter.rego that returns only workloads seeing ad sa won't be potential resources for failing, they are only used for the calculation of the control